### ISSUE TITLE
Elevate funding info

### ISSUE THREAD

**Issue by mih** [psychoinformatics-de/sfb1451-projects-catalog#67]
ATM this is a cryptic ID only. No linkage

![image](https://github.com/psychoinformatics-de/sfb1451-projects-catalog/assets/136479/cd843c0d-1505-4c38-80e2-38d6faf7a766)

The first number actually resolves to this page https://gepris.dfg.de/gepris/projekt/431549029

On that page it can be seen that B05 is actually https://gepris.dfg.de/gepris/projekt/458640473

So in some sense the latter is the real ID.

That being said, the SFB instructs its members to use the following funding acknowledgement

> SFB 1451 (431549029, B05 project)

We could consider listing both GEPRIS identifiers, and actually link the GEPRIS record for each.

**Comment by jsheunis** [comment]
This will require:
- updating the dataset schema to include identifier link(s) for a funding entity/identity
- updating the dataset page html to render the link(s)
- updating metadata records to include the link(s)

Alternatively, the current html for the rendering of the funding tab could be removed, and funding could be added to the additional display with a `@context` such that it uses the recently added mechanism for displaying semantic info. This would likely also require an html update to be able to render any new `@type`s.

@mslw what do you think?

**Comment by mslw** [comment]
I thought the same as you (with these three bullet points). I think I like Funding being its own tab. In general we would need to update the dataset html template like we did with additional tabs, plus decide on a way to feed that information to the catalog. I didn't look into the template to see how much it would have to change.

One thing to note is where this information comes from. It is added by us directly for project datasets, and there it uses all current schema fields (name, identifier, description) and looks like this - of course we can freely tweak what we provide to the catalog:

![image](https://github.com/psychoinformatics-de/sfb1451-projects-catalog/assets/11985212/d9a20952-bc03-4e32-8f34-8039ead683d5)

It can also come from tabby files, where we [asked for](https://rdm.sfb1451.de/data-catalog-submission/#table-funding-required) name DFG and identifier in the format 431549029-B05; currently shown as below but of course we can postprocess and extend it any way we like:

![image](https://github.com/psychoinformatics-de/sfb1451-projects-catalog/assets/11985212/6d209a13-2e90-4e2e-83b9-8ca2e4e18e07)

Finally, it can come from `datacite.yaml` provided by users as part of their actual DataLad datasets, there it is only available as a free-format string and we'd had to do some pattern matching (that I'm reluctant to do) to discover what is what:

![image](https://github.com/psychoinformatics-de/sfb1451-projects-catalog/assets/11985212/54881586-5920-416f-849f-e4e1c7948df6)

Bottom line: I would support updating the dataset html template to display links (if provided), and providing such links for metadata added explicitly (where we know the project name) or coming from tabby (where we can parse the pattern to look up links).

**Comment by mslw** [comment]
I believe this would be the most desired presentation using the current catalog schema (name, identifier, and description) and template:

![Screenshot from 2023-10-30 14-39-07](https://github.com/psychoinformatics-de/sfb1451-projects-catalog/assets/11985212/e51ee5ab-25cc-4cd4-8848-64ccff7ae067)

I think this covers most if not all things to display; we mostly need a way to communicate identifier as link. This opens the following questions...

### What is the identifier?

Out of the things we can put in the identifier field, what is *the* identifier?
- SFB 1451 (431549029, B05 project)
- 431549029
- https://gepris.dfg.de/gepris/projekt/431549029

### What is what?

As a reminder, the items shown in the card here, top to bottom, are "name" (of the funding source), "identifier", and "description" in catalog's schema. Thinking how to supply the information with GEPRIS identifiers (that could be displayed as links) and preferred acknowledgement to the catalog while using an existing standard, I looked into two types, [schema.org Grant](https://schema.org/Grant) and [openMINDS funding](https://openminds-documentation.readthedocs.io/en/latest/specifications/core/miscellaneous/funding.html). The latter is appealing primarily because it distinguishes acknowledgement and award number (but sadly has no URL). This is my first guess for what the properties could be, but I don't seem to have clarity and suggestions are very very welcome:

![bitmap](https://github.com/psychoinformatics-de/sfb1451-projects-catalog/assets/11985212/5bdcb439-6bf1-4a5b-9b66-6dbff8d12e28)

**Comment by mih** [comment]
I like the schema org variant. I believe it would be best to use the full gepris URL as an identifier, otherwise it is just a number and hardly unique.

**Comment by mslw** [comment]
Would you then use the whole name as link text and identifier as link target? Or show two lines, one being name as plain text, the other being identifier as a full URL (if it's a URL)?

**Comment by mslw** [comment]
Currently thinking of a slightly different name / description / alternateName assignment, leading to a card composed of:

**$** `schema:funder` -- Deutsche Forschungsgemeinschaft (DFG)
**Grant name**: `schema:name` -- SFB1451: Key mechanisms of motor control in health and disease
**Grant identifier**: `schema:identifier` -- https://gepris.dfg.de/gepris/projekt/431549029
**Alternate name**: `schema:alternateName` --  SFB 1451 (431549029, B05 project)

Possibly combining identifier and alternate name into a single link if both are present, but not sure atm.

**Comment by mslw** [comment]
That's where I'm at:

```
{
 'funding': [{'@type': 'schema:Grant',
              'alternateName': 'SFB 1451 (431549029, INF project)',
              'funder': 'Deutsche Forschungsgemeinschaft (DFG)',
              'identifier': 'https://gepris.dfg.de/gepris/projekt/431549029',
              'name': 'SFB 1451: Key mechanisms of motor control in health and '
                      'disease'},
             {'@type': 'schema:Grant',
              'funder': 'NSF-OPP',
              'identifier': '#0217282'},
...
}
```

becomes

![image](https://github.com/psychoinformatics-de/sfb1451-projects-catalog/assets/11985212/ddd0c727-9e54-4565-9966-7a65f0cadecb)

**Comment by mslw** [comment]
For the record, this is how Zenodo does it:

![image](https://github.com/psychoinformatics-de/sfb1451-projects-catalog/assets/11985212/849cea87-e069-4737-9a28-cb8a70acd2a8)

The required fields are: Funder, Number, Title, URL.

**Comment by mslw** [comment]
For this issue, it remains to merge the linked PR and re-add the tabby metadata to get the benefit of the new formatting.

**Comment by mslw** [comment]
We had a re-think of the display, and concluded that, given the dual identifier (subprojects have their own GEPRIS pages with distinct URLs, even though the subproject's page lists the main grant in the identifier field) it will make the most sense to actually show two tabs like so:

![image](https://github.com/psychoinformatics-de/sfb1451-projects-catalog/assets/11985212/19c956b9-d342-4a07-9920-66cbeb6b8bef)

I have thought about having a single card, and one of the identifiers as `sameAs`, but it isn't really this kind of relation (more like `isPartOf`, a property which is not in schema.org/Grant). We may still add `isPartOf` to the metadata object for completeness (additional properties don't invalidate it from being a `schema:Grant`) but we won't use it for display, since we aim for the display to be based on `schema:Grant`.

We agreed that the funding acknowledgement, *SFB 1451 (431549029, B03 project)*, is expected from publications, and does not need to be shown here (the two grant cards convey the same information with added detail). If needed, it could be added as `description` (seems to be the most logical property) and the template will be able to show it.

**Comment by mslw** [comment]
This has been in effect for a while - applied to all datasets in 23b51baf24e4dd847a37ab2d811b5435f894992b
